### PR TITLE
Bug 2006947: add optional e2e-aws-proxy to samples operator master branch

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

**Comment by gabemontero** [comment]
e2e-aws-proxy  will pass after https://github.com/openshift/cluster-samples-operator/pull/397 merges

**Comment by gabemontero** [comment]
hmm ....e2e-aws-proxy passed 

perhaps I have not configured the job in the same way as the failing periodic?

investigating

**Comment by gabemontero** [comment]
not seeing anything obviously wrong with the new e2e-aws-proxy job as compared to the nightly periodic

see https://bugzilla.redhat.com/show_bug.cgi?id=2006947#c3 for tldr

/assign @adambkaplan 
/assign @coreydaley 

let's get this merged and I'll start creating test PRs in builder and cluster samples operator to vet

depending on how things sort out, I'll circle back to updating non-master branches

**Comment by gabemontero** [comment]
/hold 


got assistance from @petr-muller

**Comment by gabemontero** [comment]
OK *NOW* we are getting errors that are induced by Bug 2006947

openshift/builder e2e-aws-proxy https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22168/rehearse-22168-pull-ci-openshift-builder-master-e2e-aws-proxy/1441051837330362368 is a repro, with samples bootstrapping as removed and the build tests that need the sample imagestreams are failling

**Comment by gabemontero** [comment]
/hold cancel

@adambkaplan @coreydaley  note the master/4.10 branch rehearsals will fail until the samples operator fix with this BZ merges

but I want to test that fix with the e2e-aws-proxy this PR will provide before it merges

so we need to merge this despite master/4.10 branch failures

however, the fixes to for the builder and OCM repos in releases 4.6 - 4.9 should provide suitable validation that simply switching to the proxy workflow is OK

File: ci-operator/config/openshift/builder/openshift-builder-master.yaml
Comment (adambkaplan) [inline ci-operator/config/openshift/builder/openshift-builder-master.yaml:64]: interesting - our proxy tests were not in fact running behind a proxy!

**Review by adambkaplan — APPROVED** [review]
/approve

**Comment by gabemontero** [comment]
shoot @adambkaplan it looks like the bot took your approve and added the lgtm?

the rehearsals were still cooking when this merged

I'll continue to inspect the rehearsals for any unexpected problems .... if their statuses are never reported, we probably should create some test PRs to make sure we are OK with builder/openshift-controller-manager

**Comment by gabemontero** [comment]
looks like the rehearsals were terminated when the PR merged

I'll sanity check in https://github.com/openshift/cluster-samples-operator/pull/397 followed by a test PR in build and OCM @adambkaplan

**Pull request merged**
